pocs/filterwheel/simulator.py

```diff
@@ -0,0 +1,119 @@
+    def __init__(self,
+                 name='Simulated Filter Wheel',
+                 model='simulator',
+                 camera=None,
+                 filter_names=None,
+                 timeout=10 * u.second,
+                 serial_number=None,
+                 move_time=1 * u.second,
+                 move_bidirectional=False,
+                 *args, **kwargs):
+        super().__init__(name=name,
+                         model=model,
+                         camera=camera,
+                         filter_names=filter_names,
+                         timeout=timeout,
+                         serial_number=serial_number,
+                         *args, **kwargs)
+        if isinstance(move_time, u.Quantity):
+            self._move_time = move_time.to(u.second).value
+        else:
+            self._move_time = move_time
+        self._move_birectional = bool(move_bidirectional)
+        self.connect()
+        self.logger.info("Filter wheel {} initialised".format(self))
+
+##################################################################################################
+# Properties
+##################################################################################################
+
+    @AbstractFilterWheel.position.getter
+    def position(self):
+        """ Current integer position of the filter wheel """
+        if math.isnan(self._position):
+            self.logger.warning("Filter wheel position unknown, returning NaN")
+        return self._position
+
+##################################################################################################
+# Methods
+##################################################################################################
+
+    def connect(self):
+        """ Connect to the filter wheel """
+        self._serial_number = 'SW{:04d}'.format(random.randint(0, 9999))
+        self._position = 1
+        self._moving = False
+        self.logger.info("Filter wheel {} initialised".format(self))
+        self._connected = True
+
+##################################################################################################
+# Private methods
+##################################################################################################
+
+    def _move_to(self, position, move_event):
+        if self._moving:
+            move_event.set()
+            msg = "Attempt to move filter wheel when already moving"
+            self.logger.error(msg)
+            raise RuntimeError(msg)
+
+        move_distance = position - self.position
+        if move_distance < 0:
+            if not self._move_birectional:
+                move_distance += self._n_positions
+            else:
+                move_distance = -move_distance
+        move_duration = move_distance * self._move_time
+
+        move = threading.Timer(interval=move_duration,
+                               function=self._complete_move,
+                               args=(position, move_event))
+        move.start()
```

Comment:
Just to be safe it seems like this line should go below the next two lines, since they set the condition the `move` timer changes. If for some reason the time got set to zero move time this could feasibly get stuck.

Reply:
OK. Setting the `Timer` with zero interval does seem to cause trouble, which is one of the reasons I added some code to the higher level `move_to()` method in the base class to prevent the `_move_to()` method from being called for a 'move' to the position that the filter wheel is already in.